ruff.toml

```
@@ -16,7 +16,25 @@ ignore = [
 	"ISC001",
 	"ISC002",
 ]
+extend-select = [
+	"UP",  # pyupgrade
+]
+extend-ignore = [
+	"UP015",  # redundant-open-modes, explicit is prefered
+	"UP030",  # temporarily disabled
+	"UP031",  # temporarily disabled
+	"UP032",  # temporarily disabled
+	"UP036",  # temporarily disabled
+]
+extend-exclude = [
+	"**/_vendor",
+	"setuptools/_distutils",
+]
 
 [format]
+extend-exclude = [
+	"**/_vendor",
+	"setuptools/_distutils",
```

Let's ignore this extra-code path that is automatically generated:

```suggestion
	"setuptools/_distutils",
	"setuptools/config/_validate_pyproject",
```